keyvi/src/cpp/dictionary/dictionary_compiler.h

```diff
@@ -252,11 +252,19 @@ class DictionaryCompiler
     }
 
     void Write(std::ostream& stream) {
-       generator_->Write(stream);
+      if (!generator_) {
+        throw compiler_exception("not compiled yet");
+      }
+
+      generator_->Write(stream);
     }
 
     template<typename StringType>
     void WriteToFile(StringType filename) {
+      if (!generator_) {
```

Comment:
@hendrikmuhs 
Here I suggest just to call `this->Write()` instead of `generator_->Write()`, so we have only one throw statement.